docs/tempo/website/troubleshooting/_index.md

```diff
@@ -0,0 +1,118 @@
+
+This topic helps with day zero operational issues that may come up when getting started with Tempo. It walks through debugging each part of the ingestion and query pipeline to drill down and diagnose issues. 
+
+## 1. Problem - I am unable to see any of my traces in Tempo
+
+>** Potential causes**
+- There could be issues in ingestion of the data into Tempo, that is, spans are either not being sent correctly to Tempo or they are not getting sampled.
+- There could be issues querying for traces that have been received by Tempo.
+
+### Diagnosing and fixing ingestion issues
+
+Check whether the application spans are actually reaching Tempo. The following metrics help determine this
+- `tempo_distributor_spans_received_total` 
+- `tempodb_blocklist_length`
+
+The value of both metrics should be greater than `0` within a few minutes of the application spinning up.
+You can check both metrics using -
+- The metrics page exposed from Tempo at `http://<tempo-address>:<tempo-http-port>/metrics`, or
+- In Prometheus, if it is being used to scrape metrics
+ 
+### Issue 1 - `tempo_distributor_spans_received_total` is 0
+
+If the value of `tempo_distributor_spans_received_total` is 0, possible reasons are:
+- Use of incorrect protocol/port combination while initializing the tracer in the application.
+- Tracing records not getting picked up to send to Tempo by the internal sampler.
+- Application is running inside docker and sending traces to an incorrect endpoint.
+
+#### Solutions
+
+##### Fixing protocol/port problems
+- Find out which communication protocol is being used by the application to emit traces. This is unique to every client SDK. For instance: Jaeger Golang Client uses `Thrift Compact over UDP` by default.
+- Check the list of supported protocols and their ports and ensure that the correct combination is being used. You will find the list of supported protocols and ports here: https://grafana.com/docs/tempo/latest/getting-started/#step-1-spin-up-tempo-backend
+
+##### Fixing sampling issues
+- These issues can be tricky to determine because most SDKs use a probabilistic sampler by default. This may lead to just one in a 1000 records being picked up.
+- Check the sampling configuration of the tracer being initialized in the application and make sure it has a high sampling rate.
+- Some clients also provide metrics on the number of spans reported from the application, for example `jaeger_tracer_reporter_spans_total`. Check the value of that metric if available and make sure it is greater than zero.
+- Another way to diagnose this problem would be to generate lots and lots of traces to see if some records make their way to Tempo.
+
+##### Fixing incorrect endpoint issue
+- If the application is also running inside docker, make sure the application is sending traces to the correct endpoint (`tempo:<receiver-port>`).
+
+### Issue 2 - `tempodb_blocklist_length` is 0
+
+If the value of `tempodb_blocklist_length` is 0, the possible reason is -
```

Comment:
Failed flushing and failed compactions should be combined together.  Blocklist length doesn't have much to do with querying, but both of these things indicate issues interacting with your backend.

Reply:
Yep I removed the `tempodb_blocklist_length` metric check here.